src/app/core/app.component.ts

```diff
@@ -49,6 +53,10 @@ export class AppComponent implements OnInit, OnDestroy {
     private renderer: Renderer2,
     private el: ElementRef,
   ) {
+    subWindowUserDataUpdated(window.name, () =>
+      notifyWindowOpener(window.opener as Window)
```

Comment:
1) If we can put simple html file on server and setting path (on server level) like: { domain }/notify-window-opener, then no need to load whole app of course.

2) we can inject code (with explain comment) into index.html file directly or via script, it's almost the same as 1 way

Reply:
I experimented this morning, due to cross-origin policies we have limited access to the subwindow.
If the oauth server posts a message to referer window (us) we'll be able to close the popup window. Otherwise it will require to load the app (via redirect uri) to close the popup at app startup
Or like Aleksandr said have a dedicated page of our own for that. But then I suspect that it's the redirect uri that won't work anymore since currently we can only login in the main branch and it's the same server...

Since it works with redirect uri, we could also not open a popup and instead use the redirect uri + our internal mechanism to redirect the app to a sub page (feature implemented in LTI context but not LTI-specific)

Reply:
>  If the oauth server posts a message to referer window (us) we'll be able to close the popup window. Otherwise it will require to load the app (via redirect uri) to close the popup at app startup

"oauth server posts a message to referer window " does not work due to cors limitations. The popup has to redirect to our code so that we can close it.

>  Or like Aleksandr said have a dedicated page of our own for that. But then I suspect that it's the redirect uri that won't work anymore since currently we can only login in the main branch and it's the same server...

If you mean the callback url check... there is no check on this one.

> Since it works with redirect uri, we could also not open a popup and instead use the redirect uri + our internal mechanism to redirect the app to a sub page (feature implemented in LTI context but not LTI-specific)

The request here was to do that in a popup. May be discussed but, at use, it looks more convenient to have a popup for such a usage.

Reply:
>   If we can put simple html file on server and setting path (on server level) like: { domain }/notify-window-opener, then no need to load whole app of course.

Yes, I thought this is what we discussed you should do. It will be also cleaner. But I don't know if that is easy in terms of webpack etc.

Reply:
Can a html file be added to the assets (in angular.json > architect)? That would solve the problem.

Reply:
Fixed